python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import math
+import pickle
+from types import SimpleNamespace
+from typing import Any, Dict, Mapping, Sequence, Tuple, Type, TypeVar
+
+import numpy
+
+import rmm
+
+import cudf
+from cudf.core.abc import Serializable
+from cudf.utils.string import format_bytes
+
+T = TypeVar("T", bound="Buffer")
+
+
+def cuda_array_interface_wrapper(ptr: int, size: int, owner: object = None):
```

Comment:
Maybe out of scope for this PR, but we use the `SimpleNamespace.__cuda_array_interface__` trick in multiple other places as well (mostly in `column` code IIRC) and I wonder if we could use the same helper function in multiple places. It's not immediately obvious why one would create this object so having all the places that do so call to a single, well-documented helper function could be nice. Happy to push that to a downstream refactoring PR though.

Reply:
Done